homeassistant/components/vallox/switch.py

```diff
@@ -0,0 +1,99 @@
+"""Support for Vallox ventilation unit switches."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any
+
+from vallox_websocket_api import Vallox
+
+from homeassistant.components.switch import SwitchEntity, SwitchEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ValloxDataUpdateCoordinator, ValloxEntity
+from .const import DOMAIN
+
+
+class ValloxSwitchEntity(ValloxEntity, SwitchEntity):
+    """Representation of a Vallox switch."""
+
+    entity_description: ValloxSwitchEntityDescription
+    _attr_entity_category: EntityCategory | None = EntityCategory.CONFIG
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        name: str,
+        coordinator: ValloxDataUpdateCoordinator,
+        description: ValloxSwitchEntityDescription,
+        client: Vallox,
+    ) -> None:
+        """Initialize the Vallox switch."""
+        super().__init__(name, coordinator)
+
+        self.entity_description = description
+
+        self._attr_unique_id = f"{self._device_uuid}-{description.key}"
+        self._client = client
+
+    @property
+    def is_on(self) -> bool | None:
+        """Return true if the switch is on."""
+        return self.coordinator.data.get_metric(self.entity_description.metric_key) == 1
```

Comment:
Is it correct to return `False` if no metric exists for the given key? 🤔 Should we return `None` in that case?

Reply:
Is better now.